openwisp_monitoring/db/backends/influxdb/client.py

```diff
@@ -0,0 +1,101 @@
+import logging
+from datetime import datetime
+
+from influxdb import client
+
+from .. import TIMESERIES_DB
+from .exception import DatabaseException
+
+logger = logging.getLogger(__name__)
+
+
+class DatabaseClient(DatabaseException):
+    def __init__(self):
+        self._db = None
+        self.db_name = None
+
+    def create_database(self, database=TIMESERIES_DB['NAME']):
+        """ creates database if necessary """
+        db = self.get_db()
+        # influxdb does not create a new database, neither raise an error if databese exists
+        db.create_database(database)
+        logger.info(f'Created influxdb database "{database}"')
+
+    def drop_database(self, database=TIMESERIES_DB['NAME']):
+        """ drops database if it exists """
+        db = self.get_db()
+        # influxdb does not raise an error if databese does not exist
+        db.drop_database(database)
+        logger.info(f'Dropped influxdb database "{database}"')
+
+    # TODO: Needs to be improved
+    def get_db(self):
+        """ Returns an ``InfluxDBClient`` instance """
+        if not self._db or self._db._database != self.db_name:
+            self._db = client.InfluxDBClient(
+                TIMESERIES_DB['HOST'],
+                TIMESERIES_DB['PORT'],
+                TIMESERIES_DB['USER'],
+                TIMESERIES_DB['PASSWORD'],
+                self.db_name or TIMESERIES_DB['NAME'],
+            )
+            self.db_name = self._db._database
+        return self._db
+
+    def query(self, query, **kwargs):
+        db = self.get_db()
+        database = kwargs.get('database') or self.db_name or TIMESERIES_DB['NAME']
```

Comment:
I see it too redundant, I like allowing the database parameter kwargs but `or self.db_name or TIMESERIES_DB['NAME']` seems excessive to me.

I think it's safe to remove the last `or TIMESERIES_DB['NAME']`, since self.db_name will always be defined at that point, right?